gluon-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink

```diff
@@ -92,8 +92,11 @@ if [ "$(uci get wireguard.mesh_vpn.enabled)" == "true" ] || [ "$(uci get wiregua
 		PUBLICKEY=$(uci get wireguard.mesh_vpn.privatekey | wg pubkey)
 		SEGMENT=$(uci get gluon.core.domain)
 
-		# Push public key to broker
-		gluon-wan wget -q  -O- --post-data='{"domain": "'"$SEGMENT"'","public_key": "'"$PUBLICKEY"'"}' http://broker.ffmuc.net/api/v1/wg/key/exchange
+		# Push public key to broker, try https first, fall back to http in case of error
+		gluon-wan wget -q  -O- --post-data='{"domain": "'"$SEGMENT"'","public_key": "'"$PUBLICKEY"'"}' https://broker.ffmuc.net/api/v1/wg/key/exchange
+		if [ $? -ne 0 ]; then
+			gluon-wan wget -q  -O- --post-data='{"domain": "'"$SEGMENT"'","public_key": "'"$PUBLICKEY"'"}' http://broker.ffmuc.net/api/v1/wg/key/exchange
```

Comment:
From a security perspective, this is an "insecure fallback" since an on-wire attacker could just DoS 443 and wait to eavesdrop 80. Maybe that's no big deal for this model though? Is there some other more static way you could decide if it will always use https or always use http, without needing fallback code?